src/main/java/org/opensearch/knn/plugin/transport/TrainingModelTransportAction.java

```diff
@@ -82,6 +84,7 @@ protected void doExecute(Task task, TrainingModelRequest request, ActionListener
         KNNCounter.TRAINING_REQUESTS.increment();
         ActionListener<TrainingModelResponse> wrappedListener = ActionListener.wrap(listener::onResponse, ex -> {
             KNNCounter.TRAINING_ERRORS.increment();
+            log.error("Unable to initialize model serialization for model [{}]", trainingJob.getModelId(), ex);
             listener.onFailure(ex);
```

Comment:
No, I dont think it gets logged. Here is the [RestActionListener](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/rest/action/RestActionListener.java#L86-L92) that wont log it.

Reply:
ahh thats bad. I am approving this PR, but can we just add a small comment on top of this which will state that why we are logging it here. So that in future this piece of code is not used as an example to say that we should log out the failure handler.

Reply:
@jmazanec15

Reply:
Got it. Will add.

Reply:
I think we should look into this with more details. The code reference points to a logger which is getting created like:
```
private static final Logger SUPPRESSED_ERROR_LOGGER = LogManager.getLogger("rest.suppressed");
```
Loggers generally gets configured via the string like 'rest.suppressed'. I cannot see any configuration for this `rest.suppressed` string. May be we need to check in more details, when this logger will be logged. What is the configuration for `rest.suppressed` logger.